### DISCUSSION TITLE
Clarification questions

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [Consensys/gnark#684] [Q&A]
Hi there,
I have 2 clarification questions

type CubicCircuit struct {
	X frontend.Variable `gnark:"x"`
	Y frontend.Variable `gnark:",public"`
}

1. I assume that I can't assign floats to X or Y? Instead of CubicCircuit{X: 3, Y: 35} I can't do CubicCircuit{X: 3.4, Y: 35.2} ?Currently getting this error when I assign floats "can’t set fr.Element from type float64".

2. What is the usual way to get around this in that case?

********************

type VerifyCircuit struct {
	W       [][][]frontend.Variable `gnark:"W"`
	B         [][]frontend.Variable   `gnark:"B"`
	Cx                 []frontend.Variable `gnark:",public"`
	Chidden_layer_size []frontend.Variable `gnark:"Chidden_layer_size"`
	Actcodearray       [][]frontend.Variable   `gnark:",public"`
}

3. Here the size of all the slices are to be supplied by a json.  I tried assigning []frontend.Variables to these variables. But when I do any operation on a value like Cx[1] I get  a runtime error: index out of range. Is it necessary to put the size in the struct definition?

4. If not, then how do I deal with the error?

**Comment by ivokub** [comment]
1. You cannot assign floats to variables. The SNARK arithmetic usually happens in the scalar field of the underlying elliptic curve. We have emulated emulation for non-native fields, but not for floats.
2. I think there is not good option - you can try use field emulation with a good modulus and doing fixed-point arithmetic.
3. There are two things -- when compiling the circuit we need to know the sizes of the arrays. There are two options - either use fixed-size arrays or assign already initialised slice to circuit when compiling.
